gpiozero/rotary_encoder.py

```diff
@@ -0,0 +1,253 @@
+    """
+
+    when_rotated = lambda *args: None
+
+    def __init__(self, pin_a, pin_b, pull_up=True):
+        self.pin_a = DigitalInputDevice(pin=pin_a, pull_up=pull_up)
+        self.pin_b = DigitalInputDevice(pin=pin_b, pull_up=pull_up)
+
+        self.pin_a.when_activated = self._pulse
+        self.pin_a.when_deactivated = self._pulse
+
+        self.pin_b.when_activated = self._pulse
+        self.pin_b.when_deactivated = self._pulse
+
+        self._old_a_value = self.pin_a.is_active
+        self._old_b_value = self.pin_b.is_active
+
+    def _pulse(self):
+        """
+        Calls when_rotated callback if detected changes
+        """
+        new_b_value = self.pin_b.is_active
+        new_a_value = self.pin_a.is_active
+
+        value = TableValues.value(new_b_value, new_a_value, self._old_b_value, self._old_a_value)
+
+        self._old_b_value = new_b_value
+        self._old_a_value = new_a_value
+
+        if value != 0:
+            self.when_rotated(value)
+
+    def close(self):
+        self.pin_a.close()
+        self.pin_b.close()
+
+    @property
+    def closed(self):
+        return self.pin_a.closed and self.pin_b.closed
+
+    @property
+    def is_active(self):
+        return not self.closed
+
+    @property
+    def value(self):
+        return None
+
+    def __repr__(self):
+        return "<gpiozero.%s object on pin_a %r, pin_b %r, pull_up=%s, is_active=%s>" % (
+                self.__class__.__name__, self.pin_a.pin, self.pin_b.pin, self.pin_a.pull_up, self.is_active)
+
+
+class TableValues:
+    """
+    Decodes a :class:`RotaryEncoder` pulse.
+
+               +---------+         +---------+      1
+               |         |         |         |
+     A         |         |         |         |
+               |         |         |         |
+     +---------+         +---------+         +----- 0
+
+         +---------+         +---------+            1
+         |         |         |         |
+     B   |         |         |         |
+         |         |         |         |
+     ----+         +---------+         +---------+  0
+
+    Addapted for `Paul Stoffregen table <https://github.com/PaulStoffregen/Encoder/blob/dd19b612b8050687563323777c946888f450d73c/Encoder.h#L137-L160>`_.
+    """
+
+    # The commented values are middle changes
+    values = {
+        0: +0,
+        1: +1,
+        2: -1,
+        3: +2,
+        # 4: -1,
+        5: +0,
+        6: -2,
+        # 7: +1,
+        # 8: +1,
+        9: -2,
+        10: +0,
+        # 11: -1,
+        12: +2,
+        13: -1,
+        14: +1,
+        15: +0
+    }
+
+    @staticmethod
+    def value(new_b_value, new_a_value, old_b_value, old_a_value):
+        index = TableValues.calcule_index(new_b_value, new_a_value, old_b_value, old_a_value)
+        try:
+            return TableValues.values[index]
+        except KeyError:
+            return 0
+
+    @staticmethod
+    def calcule_index(new_b_value, new_a_value, old_b_value, old_a_value):
```

Comment:
Ops! Yes. A portuguese wrong translation ("**Calcule** o índice" === "Calculate the index")